pkg/webui/console/containers/device-onboarding-form/index.js

```diff
@@ -33,50 +44,159 @@ import DeviceTypeFormSection, {
   const mayEditKeys = useSelector(state => checkFromState(mayEditApplicationDeviceKeys, state))
+  const appId = useSelector(state => selectSelectedApplicationId(state))
   const validationContext = React.useMemo(() => ({ mayEditKeys }), [mayEditKeys])
 
-  const handleSubmit = useCallback((values, formikBag, cleanedValues) => {
-    console.log(values, cleanedValues)
-    return Promise.resolve()
-  }, [])
+  const navigateToDevice = useCallback(
+    (appId, deviceId) => dispatch(push(`/applications/${appId}/devices/${deviceId}`)),
+    [dispatch],
+  )
+
+  const handleClaim = useCallback(
+    async (values, setSubmitting) => {
+      const { ids, authentication_code, qr_code } = values
+
+      let authenticatedIdentifiers
+      if (!qr_code) {
+        authenticatedIdentifiers = {
+          join_eui: ids.join_eui,
+          dev_eui: ids.dev_eui,
+          authentication_code,
+        }
+      }
+
+      try {
+        const device = await dispatch(
+          attachPromise(claimDevice(appId, qr_code, authenticatedIdentifiers)),
+        )
+        const { ids } = device
+        navigateToDevice(appId, ids.device_id)
+      } catch (error) {
+        setSubmitting(false)
+      }
+    },
+    [appId, navigateToDevice, dispatch],
+  )
+
+  const handleRegister = useCallback(
+    async (values, setSubmitting, resetForm, setErrors, cleanedValues) => {
+      try {
+        const { _registration } = values
+        const { ids, supports_join, mac_state = {} } = cleanedValues
+        ids.application_ids = { application_id: appId }
+
+        //  Do not attempt to set empty current_parameters on device creation.
+        if (
+          mac_state.current_parameters &&
+          Object.keys(mac_state.current_parameters).length === 0
+        ) {
+          delete mac_state.current_parameters
+        }
+
+        await dispatch(attachPromise(createDevice(appId, cleanedValues)))
+        switch (_registration) {
+          case REGISTRATION_TYPES.MULTIPLE:
+            toast({
+              type: toast.types.SUCCESS,
+              message: m.createSuccess,
+            })
+            resetForm({
+              errors: {},
+              values: {
+                ...values,
+                ...initialValues,
+                ids: {
+                  ...initialValues.ids,
+                  join_eui: supports_join ? values.ids.join_eui : undefined,
+                },
+                frequency_plan_id: values.frequency_plan_id,
+                version_ids: values.version_ids,
+                _registration: REGISTRATION_TYPES.MULTIPLE,
+              },
+            })
+            break
+          case REGISTRATION_TYPES.SINGLE:
+            resetForm({ values: initialValues })
+            toast({
+              type: toast.types.SUCCESS,
+              message: m.createSuccess,
+            })
+          default:
+            navigateToDevice(appId, ids.device_id)
+        }
+      } catch (error) {
+        setErrors(error)
```

Comment:
```suggestion
        setError(error)
```

`setErrors()` from the form context is used to set errors for individual fields and is not what you want here. Instead, we need a separate state `error` and `setError()` which is then passed to the form as `error` prop.